lib/config/exec.js

```diff
@@ -156,22 +151,11 @@ function exec(nodemonOptions, execMap) {
   }
 
   // allow users to make a mistake on the extension to monitor
-  // converts js,jade => js,jade
-  // and 'js jade' => js,jade
+  // converts .js, jade => js,jade
   // BIG NOTE: user can't do this: nodemon -e *.js
   // because the terminal will automatically expand the glob against
   // the file system :(
-  if (extension.indexOf(' ') !== -1 ||
-      extension.indexOf(',') !== -1 ||
-      extension.indexOf('*.') !== -1) {
-
-    extension = extension.replace(/\s+/g, '|') // convert spaces to pipes
-      .replace(/,/g, '|') // convert commas to pipes
-      .split('|') // split on those pipes
-      .map(function (item) {
-        return item.replace(/^[\*\.]+/, ''); // remove "*."
-      }).join(','); // return regexp string like: js,jade
-  }
+  extension = extension.match(/\w+/g).join(',');
```

Comment:
This was the line I didn't like, but I've run some manual tests and it's okay, but it was this change that prompted me to *want* tests. I've manually checked the examples from the comments they're fine (because the regex is fine).